File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(n, sum):
        if n==len(numbers):
            if sum==target: return 1
            else: return 0
        add=dfs(n+1, sum+numbers[n])
        sub=dfs(n+1, sum-numbers[n])
        return add+sub
    return dfs(0,0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        return dfs(numbers, target, 0, 0);
    }

    private int dfs(int[] numbers, int target, int n, int sum) {
        if (n == numbers.length) {
            if (sum == target) return 1;
            else return 0;
        }

        int add = dfs(numbers, target, n + 1, sum + numbers[n]);
        int sub = dfs(numbers, target, n + 1, sum - numbers[n]);

        return add + sub;
    }
}